File: iniciante/python/1021-notas-e-moedas.py
def cedulas(n):
    print('NOTAS:')
    notas = [100000, 50000, 20000, 10000, 5000, 2000]
    restante = n * 1000
    for nota in notas:
        quantidade_cedula = restante // nota
        restante = restante % nota
        print(f'{int(quantidade_cedula)} nota(s) de R$ {(nota/1000):.2f}')
    moedas(restante/1000)


def moedas(n):
    print('MOEDAS:')
    cents = [1000, 500, 250, 100, 50, 10]
    resto = n * 1000
    for cent in cents:
        quantidade_moeda = resto // cent
        resto = resto % cent
        print(f'{int(quantidade_moeda)} moeda(s) de R$ {(cent/1000):.2f}')


cel = float(input())
cedulas(cel)

File: iniciante/c/1021-notas-e-moedas.c
#include <stdio.h>
#include <math.h>

int main(){
    int i, nota, moeda, resto;
    int notas[6] = {10000, 5000, 2000, 1000, 500, 200};
    int moedas[6] = {100, 50, 25, 10, 5, 1};
    float entrada;

    scanf("%f", &entrada);
    resto = lround(100.0 * entrada);

    printf("NOTAS:\n");
    for(i = 0; i < 6; i++){
        nota = resto / notas[i];
        resto = resto % notas[i];
        printf("%d nota(s) de R$ %d.00\n", nota, notas[i] / 100);
    }

    printf("MOEDAS:\n");
    for(i = 0; i < 6; i++){
        moeda = resto / moedas[i];
        resto = resto % moedas[i];
        printf("%d moeda(s) de R$ %.2f\n", moeda, (float) moedas[i] / 100);
    }

    return 0;

}